tests/foreman/api/test_computeresource_gce.py

```diff
@@ -0,0 +1,357 @@
+        assert len(satgce_images) == len(gcloudclinet_images)
+
+    @tier3
+    def test_positive_check_available_networks(self, module_gce_compute, googleclient):
+        """Verify all the networks from GCE are available to select
+
+        :id: 259c9f5f-19dd-4e06-af5e-874612914f44
+
+        :expectedresults: All the networks from Google CR should be available to select in GCE CR
+
+        :CaseLevel: Integration
+        """
+        satgce_networks = module_gce_compute.available_networks()['total']
+        gcloudclient_networks = googleclient.list_network()
+        assert int(satgce_networks) == len(gcloudclient_networks)
+
+    @pytest.mark.skip_if_open("BZ:1794744")
+    @tier3
+    def test_positive_check_available_flavors(self, module_gce_compute):
+        """Verify flavors from GCE are available to select
+
+        :id: f1da6706-9c27-4730-98e2-f710b847ff97
+
+        :BZ: 1794744
+
+        :expectedresults: All the flavors from Google CR should be available to select in GCE Host
+
+        :CaseLevel: Integration
+        """
+        satgce_flavors = int(module_gce_compute.available_flavors()['total'])
+        assert satgce_flavors > 1
+
+    @tier3
+    def test_positive_create_finish_template_image(
+            self, module_gce_finishimg, module_gce_compute, gce_latest_rhel_uuid):
+        """Finish template image along with username is being added in GCE CR
+
+        :id: ec7ac618-ed1d-4b36-af7b-995cf4f78341
+
+        :BZ: 1794744
+
+        :steps:
+            1. Create a GCE Compute Resource.
+            2. Add a finish template based image in GCE CR.
+
+        :expectedresults: Finish template image should be added in GCE CR along with username
+
+        :CaseLevel: Integration
+        """
+        assert module_gce_finishimg.compute_resource.id == module_gce_compute.id
+        assert module_gce_finishimg.uuid == gce_latest_rhel_uuid
+        assert module_gce_finishimg.username == finishuser
+
+    @tier3
+    def test_positive_create_cloud_init_image(
+            self, module_gce_cloudimg, module_gce_compute, gce_custom_cloudinit_uuid):
+        """Cloud Init template image along with username is being added in GCE CR
+
+        :id: 5a30c227-324a-49f9-932f-ce80da8611d0
+
+        :steps:
+            1. Create a GCE Compute Resource.
+            2. Add a cloud init template in GCE CR.
+
+        :expectedresults: Cloud init image should be added in GCE CR along with username
+
+        :CaseLevel: Integration
+        """
+        assert module_gce_cloudimg.compute_resource.id == module_gce_compute.id
+        assert module_gce_cloudimg.uuid == gce_custom_cloudinit_uuid
+        assert module_gce_cloudimg.username == clouduser
+
+
+class TestGCEHostProvisioningTestCase:
+    """GCE Host Provisioning Tests
+
+    :Requirement: Host
+
+    :CaseComponent: Hosts
+    """
+    @pytest.fixture(scope='class')
+    def gce_domain(self, module_org, module_location, module_smart_proxy):
```

Comment:
Can you move these fixtures to top of file? Usually fixtures are defined as global functions, not as methods of class. having them here is unexpected.

Reply:
Not mandatory to Keep in Module Scope but yes the newly added tests (by new contributor) for that person/tests it will be helpful.

Moving at module level.